Cauldron/Assets/Editor/SODictionaryLoader.cs

```diff
@@ -5,37 +5,52 @@
 namespace Editor
 {
     [InitializeOnLoad]
-    public class SODictionaryLoader
+    [CustomEditor(typeof(SODictionary))]
+    public class SODictionaryLoader : UnityEditor.Editor
     {
+        private static SODictionary targetDictionary;
         static SODictionaryLoader()
         {
             EditorApplication.projectChanged += LoadDictionary;
         }
 
-        static void LoadDictionary()
+        public override void OnInspectorGUI()
         {
-            var dictionary = ScriptableObjectHelper.LoadSingleAsset<SODictionary>();
+            if (GUILayout.Button("Update SODictionary", GUILayout.Height(40)))
+            {
+                targetDictionary = ScriptableObjectHelper.LoadSingleAsset<SODictionary>();
```

Comment:
This is a custom editor class for a ScriptableObject, so it is perfectly capable to get that object without loading it externally.
Use 'target' field and a cast, see other custom editor classes in the project.